src/TesApi.Web/TesEventMessage.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using TesApi.Web.Storage;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Represents the events sent by the node task runner.
+    /// </summary>
+    /// <remarks>This should be transient in DI.</remarks>
+    public class TesEventMessage
+    {
+        static TesEventMessage() => Tes.Utilities.NewtonsoftJsonSafeInit.SetDefaultSettings();
+
+        /// <summary>
+        /// Blob tag used to record event processing.
+        /// </summary>
+        public const string ProcessedTag = "processed";
+
+        private readonly IStorageAccessProvider _storageAccessProvider;
```

Comment:
This notation is not used for the most part in the code base. We should avoid it for consistency's sake, even if it is better than the current style/pattern.

Reply:
Working on that...